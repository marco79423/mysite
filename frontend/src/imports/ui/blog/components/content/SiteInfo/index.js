import SiteInfoContainer from './SiteInfo'

export default SiteInfoContainer
