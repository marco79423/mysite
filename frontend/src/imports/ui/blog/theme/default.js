const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}