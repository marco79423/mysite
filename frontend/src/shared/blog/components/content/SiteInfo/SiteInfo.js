import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import {Table, Tbody, Tr, Td} from '../../generic/table'
import TitleLink from '../../generic/TitleLink'

const Base = styled.section`
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    min-height: 600px;
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;
  }
`

const Header = styled.header`
  h1 {
    margin: 3px 0 24px;

    a {
      font-size: 2rem;
    }
  }
`

export default class SiteInfo extends React.PureComponent {
  static PropTypes = {
    repositoryVersion: PropTypes.string.isRequired,
    updatedTime: PropTypes.string.isRequired
  }

  renderHeader = () => {
    return (
      <Header>
        <h1><TitleLink to='/info/'>網站資訊</TitleLink></h1>
      </Header>
    )
  }

  renderInfoTable = () => {
    return (
      <Table>
        <Tbody>
        <Tr>
          <Td>網站更新時間：</Td>
          <Td>{this.props.updatedTime}</Td>
        </Tr>
        <Tr>
          <Td>儲存庫版本：</Td>
          <Td>{this.props.repositoryVersion}</Td>
        </Tr>
        </Tbody>
      </Table>
    )
  }

  render () {
    return (
      <Base>
        <article>
          {this.renderHeader()}
          {this.renderInfoTable()}
        </article>
      </Base>
    )
  }
}
