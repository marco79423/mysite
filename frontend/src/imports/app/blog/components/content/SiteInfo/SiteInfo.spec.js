import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from 'styled-components'

import SiteInfo from './SiteInfo'
import theme from '../../../theme/default'


test('It should render SiteInfo component correctly', () => {
  const props = Immutable.fromJS({
    frontendVersion: 'master (3f5eb304b501ea2b5437ed6c4612bbc66077b7b8)',
    backendVersion: 'develop (81ccde3550325c06a10b6acce75b4df529955472)',
    updatedTime: '2017-04-08 12:01:57.198000'
  })

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <SiteInfo
        frontendVersion={props.get('frontendVersion')}
        backendVersion={props.get('backendVersion')}
        updatedTime={props.get('updatedTime')}
      />
    </ThemeProvider>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
