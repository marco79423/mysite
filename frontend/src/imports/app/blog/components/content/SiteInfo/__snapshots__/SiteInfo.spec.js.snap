// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render SiteInfo component correctly 1`] = `
.c4 {
  border-collapse: collapse;
  width: 100%;
  font-size: 1.2em;
}

.c5 {
  background: #eee;
}

.c5:nth-of-type(even) {
  background: #f9f9f9;
}

.c6 {
  padding: 0.6em;
  border: 1px solid #ccc;
}

.c3 {
  color: #0097A7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  color: #006064;
}

.c2 {
  font-size: 2rem;
}

.c0 {
  float: left;
  width: 800px;
}

.c0 article {
  min-height: 600px;
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;
}

.c1 h1 {
  margin: 3px 0 24px;
}

.c1 h1 a {
  font-size: 2rem;
}

@media (max-width:1200px) {
  .c0 {
    width: 100%;
  }
}

<section
  className="c0"
>
  <article>
    <header
      className="c1"
    >
      <h1>
        <a
          className="c2 c3"
          onClick={[Function]}
          style={Object {}}
        >
          網站資訊
        </a>
      </h1>
    </header>
    <table
      className="c4"
    >
      <tbody
        className=""
      >
        <tr
          className="c5"
        >
          <td
            className="c6"
          >
            網站更新時間：
          </td>
          <td
            className="c6"
          >
            2017-04-08 12:01:57.198000
          </td>
        </tr>
        <tr
          className="c5"
        >
          <td
            className="c6"
          >
            儲存庫版本：
          </td>
          <td
            className="c6"
          />
        </tr>
      </tbody>
    </table>
  </article>
</section>
`;
